src/components/table/ResourceName.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { createStyles, WithStyles, withStyles } from '@mui/styles';
+import { Link } from 'react-router-dom';
+import { CustomTheme } from '@mui/material/styles';
+import { IconButton } from '@mui/material';
+import { PersonAdd } from '@mui/icons-material';
+import { DatasetSummaryModel, SnapshotSummaryModel } from 'generated/tdr';
+import _ from 'lodash';
+import TerraTooltip from 'components/common/TerraTooltip';
+
+import { ResourceType } from '../../constants';
+
+const styles = (theme: CustomTheme) =>
+  createStyles({
+    nameWrapper: {
+      display: 'flex',
+    },
+    // eslint-disable-next-line prefer-object-spread
+    jadeLinkWrapper: Object.assign({ flexGrow: 1 }, theme.mixins.ellipsis),
+    jadeLink: {
+      ...theme.mixins.jadeLink,
+    },
+    addAsSteward: {
+      padding: 0,
+      '& svg': {
+        height: 20,
+        fill: theme.palette.primary.main,
+      },
+    },
+  });
+
+function hasAdminOnlyAccess(id: string, roleMaps: { [key: string]: Array<string> }) {
+  const roles = roleMaps[id];
+  return _.isArray(roles) && roles.length === 1 && roles[0] === 'admin';
```

Comment:
Any reason to use _.isArray instead of Array.isArray?

https://github.com/lodash/lodash/blob/ddfd9b11a0126db2302cb70ec9973b66baec0975/lodash.js#L11286

Reply:
Mostly just that I'm already familiar with lodash :) I'm not sure if there's a benefit of one over the other.  I like the idea of lodash since there's the one entry point to do all of the things

Reply:
I'll save you the link trip.  _.isArray just calls Array.isArray.

Reply:
Haha, I may just leave it (since I like having the facade with lodash and it's not any more complicated)

Reply:
After playing with this locally for a bit would you consider a slight tweak to the logic?
return _.isArray(roles) && roles.includes('admin') && !roles.includes('steward');

Handles the case where you remove yourself as a steward, remain a custodian or reader, and want to let yourself back as a steward.

Reply:
Yeah, I was debating. I guess I like the simplicity of it now and it has the added benefit of working for both snapshots and datasets (datasets and snapshots have steward and admin but after that they diverge) so if it's ok, I'd like to keep it as is